File: leetcode/[20]ValidParentheses.js
// try 1
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const obj = {
        "(" :")",
        "{" :"}",
        "[" :"]",
    };
    
    // console.log(Object.values(obj))
    for (let i = 1; i < s.length; i+=2) {
        console.log(s[i - 1]);
        let t = s[i - 1];
        console.log(obj[t]);
        // console.log(Object.values(obj)[i]);
        if (obj[t] !== Object.values(obj)[i]) {
            return false;
        }
    }
    return true;
};

File: leetcode/[20]ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        brackets = {
            "]": "[",
            "}": "{",
            ")": "(",
        }

        for bracket in s:
            if bracket in brackets:
                top_element = stack.pop() if stack else "#"
                if brackets[bracket] != top_element:
                    return False
            else:
                stack.append(bracket)

        return not stack